deephyper/skopt/optimizer/optimizer.py

```diff
@@ -651,10 +673,12 @@ def _filter_failures(self, yi):
         if self.filter_failures in ["mean", "max"]:
             yi_no_failure = [v for v in yi if v != OBJECTIVE_VALUE_FAILURE]
 
-            if self.filter_failures == "mean":
-                yi_failed_value = np.mean(yi_no_failure)
+            if not yi_no_failure:
```

Comment:
`not` of a list is ambiguous replace with `if len(yi_no_failure) == 0` seems better or add a comment such as `# case where no valid objective is found`

The logic of this code is not clear, if all returned objectives are failures, you fit the estimator with a list of `np.nan` values? Maybe a new randomly sampled configuration should be returned.

```python
X = self._ask_random_points(size=n_points)
self.sampled.extend(X)
return X
```

Reply:
I was just trying to follow the previous behavior with the single-objective case. In the single-objective case also it is possible that that all past evaluations are failures. Not sure how this is handled currently. I can try to replicate that for the multi-objective case

Reply:
Then we can probably do this for Bose case

```python
yi_no_failure = [v for v in yi if v != OBJECTIVE_VALUE_FAILURE]

if len(yi_no_failure) == 0:
  X = self._ask_random_points(size=n_points)
  self.sampled.extend(X)
  return X

if self.filter_failures == "mean":
  ...
```

Reply:
Sounds good